mart/attack/adversary.py

```diff
@@ -150,35 +166,32 @@ def configure_gradient_clipping(
                 self.gradient_modifier(group["params"])
 
     @silent()
-    def forward(self, *, model=None, **batch):
-        batch["model"] = model
+    def forward(self, *, batch: torch.Tensor | list | dict, model: Callable):
+        input, target = batch
 
-        # Adversary can live within a sequence of model. To signal the adversary should
-        # attack, one must pass a model to attack when calling the adversary. Since we
-        # do not know where the Adversary lives inside the model, we also need the
-        # remaining sequence to be able to get a loss.
```

Comment:
Okay so we're getting rid of this functionality?

Reply:
Yes. The new interface is simpler and it works for Anomalib and Armory.